dart/lib/src/hub.dart

```diff
@@ -420,14 +420,25 @@ class Hub {
         "Tracing is disabled and this 'getSpan' returns null.",
       );
     } else {
-      final item = _peek();
+      span = Zone.current[#_span] as ISentrySpan?;
```

Comment:
l: I wouldn't use `Symbol ` as key since `Symbol` are more related to reflection. Also I don't believe I've seen it being used this way on the SDK.

Reply:
This is not reflection, that's just how you add something to a zone. https://dart.dev/articles/archive/zones#storing-zone-local-values

But the key should probably have a `sentry` prefix, as the keys are global for a zone.

Reply:
Prefixing a [symbol literal with `_`](https://api.dart.dev/stable/2.15.1/dart-core/Symbol/Symbol.html) makes them file private, just like other identifiers.

Reply:
Huh, TIL. I thought that if any other code also adds a `_span` identifier, this one is overridden because it's a key after all?

Reply:
Private symbol literals create unique symbols in each file in which they are used, so `#_span` in file `a.dart` is not the same symbol as `#_span` in file `b.dart`. If those symbols are used as keys in a map, they won't override each other since they have unique identities.

Reply:
Thanks for the explanation

Reply:
@ueman I understand that there is no reflection involved. But the key could be any object that implements `==`  and `hashCode`.

In the `Symbol` [documentation](https://api.dart.dev/stable/2.15.1/dart-core/Symbol-class.html) you can see that they are related to reflections. Not that they could not be used in other places, but that is their purpose. But I like the uniqueness by file that @blaugold mentioned.